### PR TITLE
Add dwave-preprocessing and update some packages

### PR DESCRIPTION
**Pull request by hhtong**

### CODE REVIEW

**Comment by randomir** [comment]
@hhtong, can you please split this PR in two:
1) drop support for py35 (close #91)
2) update packages as above

Thanks!

**Comment by randomir** [comment]
Would you mind rebasing now that #130 is merged? Thanks.

**Comment by hhtong** [comment]
> Would you mind rebasing now that #130 is merged? Thanks.

Done!

File: tests/test_individually.py [hunk 1]
```diff
@@ -90,3 +90,14 @@ def test_neal(self):
...
 
         resp = neal.SimulatedAnnealingSampler().sample_ising(h, J)
+
+    def test_preprocessing(self):
+        import dimod
+        from dwave.preprocessing.lower_bounds import roof_duality
+        from dwave.preprocessing.composites import ScaleComposite
+
+        bqm = dimod.BinaryQuadraticModel.from_ising({'a': 10}, {'ab': -1, 'bc': 1})
+        result = roof_duality(bqm)
```
Comment (randomir) [inline tests/test_individually.py:100]: How about a sanity check as well?
```suggestion
        result = roof_duality(bqm)
        self.assertFalse(bqm.variables - result)
```

File: tests/test_individually.py [hunk 2]
```diff
@@ -90,3 +90,14 @@ def test_neal(self):
...
+
+        sampler = ScaleComposite(dimod.ExactSolver())
+        result = sampler.sample(bqm)
```
Comment (randomir) [inline tests/test_individually.py:103]:
```suggestion
        result = sampler.sample(bqm, scalar=0.5)
        self.assertIn('scalar', result.info)
```

**Review by randomir — APPROVED** [review]
LGTM!

Ideally, we can fix the docs in this PR as well.

**Pull request merged**